### ISSUE TITLE
Blacklist feature restrictive, usage completely unclear

### ISSUE THREAD

**Issue by neilalexander** [yacy/yacy_search_server#346]
I am having a hard time getting my blacklist to recognise and filter URLs such as:

- `http://1.2.3.4/`
- `http://[2a00::1]/`
- `http://1.2.3.4:8080/`
- `http://[2a00::1]:8080/`

Similarly, I also can't get the blacklist to recognise cases where the domain part is unknown but the path is, e.g.

- `http://any-domain-here/ipfs/`

Do blacklists *really* only work for a very small subset of cases where both a domain name and path element are provided?

**Comment by Orbiter** [comment]
yes, you are completely true, we (I!) implemented something which is not so good.
The pattern had to have a * inside, either as subdomain wildcard or as tld wildcard. I believed this would cause a better option to index the host pattern inside a hashtable which would compute better for many thousands of entries. I designed this 2005 to omit an iterative testing of regular expressions.

We should do a redesign here. I admit that I am lacking time to do this. Help wanted!
